fbfmaproom/fbfmaproom.py

```diff
@@ -69,16 +72,34 @@
 
 APP.layout = fbflayout.app_layout()
 
-
-def table_columns(obs_config, obs_dataset_key):
-    obs_dataset_names = {k: v["label"] for k, v in obs_config.items()}
-    tcs = [
-        dict(id="year_label", name="Year"),
-        dict(id="enso_state", name="ENSO State"),
-        dict(id="forecast", name="Forecast, %"),
-        dict(id="obs_rank", name=f"{obs_dataset_names[obs_dataset_key]} Rank"),
-        dict(id="bad_year", name="Reported Bad Years"),
-    ]
+def table_columns(obs_dsets, obs_state):
+    obs_dataset_names = {k: v["label"] for k, v in obs_dsets.items()}
+    tcs = OrderedDict()
+    tcs["year_label"] = dict(name="Year", dynamic=None, style=None,
+                             tooltip="The year whose forecast is displayed on the map")
+    tcs["enso_state"] = dict(name="ENSO State", dynamic=None,
+                             tooltip="Displays whether an El Niño, Neutral, or La Niña state occurred during the year",
+                             style=lambda row: {'El Niño': 'cell-el-nino',
+                                                'La Niña': 'cell-la-nina',
+                                                'Neutral': 'cell-neutral'}.get(row['enso_state'], ""))
+    tcs["forecast"] = dict(name="Forecast, %", dynamic=None,
+                           tooltip="Displays all the historical flexible forecast for the selected issue month and location",
+                           style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_pnep'] == 1 else "")
+    tcs["obs_rank"] = dict(name=f"{obs_dataset_names[obs_state[0]]} Rank",
+                           style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_obs'] == 1 else "",
+                           tooltip=None,
+                           dynamic=dict(type='obs_rank',
+                                        options=obs_dataset_names,
+                                        value=obs_state[0]))
+    # for i, k in enumerate(obs_state):
+    #     tcs["obs_rank" + str(i)] = dict(name=f"{obs_dataset_names[k]} Rank",
+    #                                     dynamic=dict(type='obs_rank',
+    #                                                  options=obs_dataset_names,
+    #                                                  value=k))
+    tcs["bad_year"] = dict(name="Reported Bad Years", dynamic=None,
+                           style=lambda row: "cell-" + {'Bad': 'bad', '': 'good'}[row['bad_year']] + "-year" \
+                                              if pd.notna(row['bad_year']) else "",
```

Comment:
This is still essentially two nested ternary operators, just using even more opaque syntax for the first one. I think it would be a lot easier to read if it were written as if/elif/else.

Just expressing an opinion; not an urgent problem that needs to be addressed before we move on.